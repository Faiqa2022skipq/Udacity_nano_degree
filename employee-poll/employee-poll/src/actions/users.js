export const RECEIVE_USERS = "RECEIVE_USERS";
export const ADD_ANSWER_U = "ADD_ANSWER_U";
export const ADD_QUESTION_U = "ADD_QUESTION_U";

export function receiveUsers(users) {
    return {
        type: RECEIVE_USERS,
        users,
    };
}

export function addAnswerUser(authedUser, qid, answer) {
    return {
        type: ADD_ANSWER_U,
        authedUser,
        qid,
        answer,
    };
}

export function addQuestionUser({author, id}) {
    return {
        type: ADD_QUESTION_U,
        author,
        qid: id,
    };
}
