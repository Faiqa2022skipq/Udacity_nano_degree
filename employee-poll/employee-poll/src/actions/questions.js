import {_saveQuestion1, _saveQuestionAnswer1} from "../Services/Data";
import {addAnswerUser, addQuestionUser} from "./users";

export const ADD_QUESTION = "ADD_QUESTION";
export const ADD_ANSWER_QUESTION = "ADD_ANSWER_QUESTION";
export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";

export function receiveQuestions(questions) {
    return {
        type: RECEIVE_QUESTIONS,
        questions,
    };
}

function addQuestion(question) {
    return {
        type: ADD_QUESTION,
        question,
    };
}

function addAnswerQuestion(author, qid, answer) {
    return {
        type: ADD_ANSWER_QUESTION,
        author,
        qid,
        answer,
    };
}

export function handleAddQuestion(firstOption, secondOption) {
    return (dispatch, getState) => {
        const { authedUser } = getState();

        return _saveQuestion1(firstOption, secondOption, authedUser)
            .then((question) => {
                dispatch(addQuestion(question));
                dispatch(addQuestionUser(question))
                alert("Successfully saved!")
            })
            .catch((error) =>{
                alert("error while saving ", error)
            })
    };
}

export function handleAddAnswer(questionId, answer) {
    return (dispatch, getState) => {
        const { authedUser } = getState();
        return _saveQuestionAnswer1(authedUser.id, questionId, answer)
            .then(() => {
                dispatch(addAnswerQuestion(authedUser.id, questionId, answer));
                dispatch(addAnswerUser(authedUser.id, questionId, answer));
            });
    };
}
