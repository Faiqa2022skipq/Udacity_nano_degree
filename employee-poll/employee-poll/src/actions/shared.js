import {receiveUsers} from "./users";
import {receiveQuestions} from "./questions";
import {getInitialData} from "../Services/Data"

export function handleInitialData() {
    return (dispatch) => {
        return getInitialData().then(({users, questions}) => {
            dispatch(receiveUsers(users));
            dispatch(receiveQuestions(questions));
        });
    };
}


