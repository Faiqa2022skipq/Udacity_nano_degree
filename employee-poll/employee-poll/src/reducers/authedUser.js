import {LOGOUT_USER, AUTHED_USER} from "../actions/authedUser";

export default function authedUser(state = null, action) {
  switch (action.type) {
    case AUTHED_USER:
      return action.authedUser;
    case LOGOUT_USER:
      return null;
    default:
      return state;
  }
}
