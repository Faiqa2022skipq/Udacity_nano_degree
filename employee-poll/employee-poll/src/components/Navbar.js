import React from "react";
import {
  Nav,
  NavLink,
  NavMenu,
  NavBtn,
  NavBtnLink,
} from "./NavbarElement";
import { connect } from "react-redux";
import { useState } from "react";
import { handleLogout } from "../actions/authedUser";

const Navbar = ({ dispatch, authedUserId, authedUserAvatar }) => {
  const [isHovering, setIsHovering] = useState(false);

  const handleMouseOver = () => {
    setIsHovering(true);
  };

  const handleMouseOut = () => {
    setIsHovering(false);
  };


  return (
    <>

      <Nav>
        <div style={{ marginLeft: '250px' }}>


          <NavMenu>
            <NavLink
              to="/"
              activestyle={{ color: 'black' }}
              data-testid="home"
            >
              Home
            </NavLink>
            <NavLink
              to="/leadership"
              activestyle={{ color: 'black' }}
              data-testid="leadership"

            >
              Leaderboard
            </NavLink>
            <NavLink
              to="/NewPoll"
              activestyle={{ color: 'black' }}
              data-testid="new"
            >
              New Poll
            </NavLink>
            <span style={{ color: 'white' }} onMouseOver={handleMouseOver} onMouseOut={handleMouseOut}

            >  {<img src={authedUserAvatar} alt="avatar" style={{
              width: '30px',
              height: '20%',
              margin: 7,
              borderRadius: 7

            }} />}
              {isHovering && ` ${authedUserId}`}


            </span>

            <NavBtn>



              <NavBtnLink to="/login" onClick={((e) => { dispatch(handleLogout(e)) })} activestyle={authedUserId} data-testid="logout" style={{ float: 'right' }}>Logout</NavBtnLink>
            </NavBtn>
          </NavMenu>
        </div>
      </Nav>

    </>
  );
};
const mapStateToProps = ({ authedUser }) => ({
  authedUserId: authedUser.id,
  authedUserAvatar: authedUser.avatarURL,
});
export default connect(mapStateToProps)(Navbar);