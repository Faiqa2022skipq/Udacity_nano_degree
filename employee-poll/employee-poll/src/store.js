
import middleware from './middleware';
import reducers from './reducers';
import { createStore } from '@reduxjs/toolkit';
export const store = createStore(
 reducers,
middleware
);
