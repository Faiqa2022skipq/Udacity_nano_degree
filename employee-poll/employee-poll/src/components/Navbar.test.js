import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {store} from "../store";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import Navbar from "./Navbar";
import {setAuthedUser} from "../actions/authedUser";

describe("Navbar", () => {
    it("should render the component", () => {
        store.dispatch(setAuthedUser({id: "tylermcginnis", password: ""}));

        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Navbar/>
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
        const nav_leadership = component.getByTestId('leadership')
        expect(nav_leadership).toBeInTheDocument()
        const nav_home = component.getByTestId('home')
        expect(nav_home).toBeInTheDocument()
        const nav_new = component.getByTestId('new')
        expect(nav_new).toBeInTheDocument()
        const nav_logout = component.getByTestId('logout')
        expect(nav_logout).toBeInTheDocument()
    });
})
