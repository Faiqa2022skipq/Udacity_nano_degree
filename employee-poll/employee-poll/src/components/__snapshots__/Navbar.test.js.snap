// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-bczRLJ iA-dZku"
      >
        <div
          style="margin-left: 250px;"
        >
          <div
            class="sc-eCYdqJ xjNhp"
          >
            <a
              activestyle="[object Object]"
              aria-current="page"
              class="sc-dkzDqf jyuNrD active"
              data-testid="home"
              href="/"
            >
              Home
            </a>
            <a
              activestyle="[object Object]"
              class="sc-dkzDqf jyuNrD"
              data-testid="leadership"
              href="/leadership"
            >
              Leaderboard
            </a>
            <a
              activestyle="[object Object]"
              class="sc-dkzDqf jyuNrD"
              data-testid="new"
              href="/NewPoll"
            >
              New Poll
            </a>
            <span
              style="color: white;"
            >
                
              <img
                alt="avatar"
                style="width: 30px; height: 20%; margin: 7px; border-radius: 7px;"
              />
            </span>
            <nav
              class="sc-jSMfEi elSapS"
            >
              <a
                activestyle="tylermcginnis"
                class="sc-gKXOVf kxQgf"
                data-testid="logout"
                href="/login"
                style="float: right;"
              >
                Logout
              </a>
            </nav>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-bczRLJ iA-dZku"
    >
      <div
        style="margin-left: 250px;"
      >
        <div
          class="sc-eCYdqJ xjNhp"
        >
          <a
            activestyle="[object Object]"
            aria-current="page"
            class="sc-dkzDqf jyuNrD active"
            data-testid="home"
            href="/"
          >
            Home
          </a>
          <a
            activestyle="[object Object]"
            class="sc-dkzDqf jyuNrD"
            data-testid="leadership"
            href="/leadership"
          >
            Leaderboard
          </a>
          <a
            activestyle="[object Object]"
            class="sc-dkzDqf jyuNrD"
            data-testid="new"
            href="/NewPoll"
          >
            New Poll
          </a>
          <span
            style="color: white;"
          >
              
            <img
              alt="avatar"
              style="width: 30px; height: 20%; margin: 7px; border-radius: 7px;"
            />
          </span>
          <nav
            class="sc-jSMfEi elSapS"
          >
            <a
              activestyle="tylermcginnis"
              class="sc-gKXOVf kxQgf"
              data-testid="logout"
              href="/login"
              style="float: right;"
            >
              Logout
            </a>
          </nav>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
